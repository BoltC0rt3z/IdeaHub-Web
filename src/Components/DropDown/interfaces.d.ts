export interface DropDownItem {
  name: string;
  link?: string;
}
