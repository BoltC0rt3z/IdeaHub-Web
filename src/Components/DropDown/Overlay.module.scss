.overlayContainer {
  margin-top: 12px;
}
