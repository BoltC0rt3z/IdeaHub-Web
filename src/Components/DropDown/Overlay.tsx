import React from 'react';
import { Dropdown } from '@primer/components';

import styles from './Overlay.module.scss';

interface Props {
  className?: string;
}

const Overlay: React.FC<Props> = (props) => {
  const { className } = props;
  const classNames: string = [styles.overlayContainer, className].join(' ');

  return (
    <div className={classNames}>
      <Dropdown.Menu>
        <Dropdown.Item>Item 1</Dropdown.Item>
      </Dropdown.Menu>
    </div>
  );
};

export default Overlay;
