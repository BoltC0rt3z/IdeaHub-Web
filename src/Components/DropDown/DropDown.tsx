import React from 'react';
import { Dropdown as AntDropdown } from 'antd';

import Overlay from './Overlay';

const DropDown: React.FC = (props) => {
  const { children } = props;

  return (
    <AntDropdown trigger={['click']} overlay={<Overlay />}>
      {children}
    </AntDropdown>
  );
};

export default DropDown;
