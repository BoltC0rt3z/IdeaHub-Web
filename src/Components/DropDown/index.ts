import DropDown from './DropDown';
import Overlay from './Overlay';

export { DropDown, Overlay };
