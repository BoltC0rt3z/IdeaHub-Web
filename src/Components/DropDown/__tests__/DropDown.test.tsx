import React from 'react';
import { mount, ReactWrapper } from 'enzyme';
import DropDown from '../DropDown';

describe('Tests for DropDown component', () => {
  const wrapper: ReactWrapper = mount(
    <DropDown>
      <div>Child</div>
    </DropDown>,
  );

  it('Should render dropdown without crushing', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
