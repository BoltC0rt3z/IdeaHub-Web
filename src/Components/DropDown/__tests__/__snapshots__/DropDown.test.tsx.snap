// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for DropDown component Should render dropdown without crushing 1`] = `ReactWrapper {}`;
